core/src/test/java/com/github/mizool/core/TestOptionals.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2019 incub8 Software Labs GmbH
+ * Copyright 2019 protel Hotelsoftware GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.mizool.core;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+
+public class TestOptionals
+{
+    @Test(dataProvider = "optionalStreams")
+    public <T> void testMapToValue(Collection<Optional<T>> input, Iterable<T> expected)
+    {
+        List<T> actual = input.stream().flatMap(Optionals::mapToValue).collect(Collectors.toList());
+        assertThat(actual).containsExactlyElementsOf(expected);
```

Comment:
Good point!

Also, to prevent readers from thinking "ah, it's a list, so order is relevant", I switched from `List` to `Set`. This is just a hint, though, because `ImmutableSet` also makes guarantees about iteration order.